src/driver/driver.rs

```diff
@@ -0,0 +1,45 @@
+//! The driver trait which all drivers will implement.
+
+use schema::Definition;
+use error::{Error, ErrorCode};
+use patch::Patch;
+use value::{Pointer, Value};
+use query::Query;
+
+pub trait Driver {
+  /// Connects to a driver and returns a driver instance. After calling this
+  /// the driver is ready to roll!
+  fn connect(uri: &str) -> Result<Box<Self>, Error>;
+
+  /// Validates an Ardite Schema Definition dependending on the driver’s
+  /// contracts with the developer. Note that the definition object will not be
+  /// associated with the driver object. This is intentional, the driver should
+  /// not be able to influence the definition without an external tool.
+  fn validate_definition(_: &Definition) -> Result<(), Error> {
```

Comment:
By default there are no validations on the definition. If a driver needs the definition to be in a certain format it will override this method and perform the validations it needs.